src/material/icon/testing/noop-icon-registry.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {NgModule, OnDestroy} from '@angular/core';
+import {MatIconRegistry} from '@angular/material/icon';
+import {Observable, of as observableOf} from 'rxjs';
+
+// tslint:disable:no-any Impossible to tell param types.
+type PublicApi<T> = {
+  [K in keyof T]: T[K] extends (...x: any[]) => T ? (...x: any[]) => PublicApi<T> : T[K]
+};
+// tslint:enable:no-any
+
+/**
+ * A null icon registry that must be imported to allow disabling of custom
+ * icons.
+ */
+export class NoopMatIconRegistry implements PublicApi<MatIconRegistry>, OnDestroy {
```

Comment:
I think that this should be marked as `Injectable`, similar to the real icon registry.

Reply:
It appears that's necessary with Ivy, yes. On the old engine it wasn't necessary to tag service with `@Injectable` if they had no dependency and they were provided with an `@NgModule` as a provider. So I will update that.